### PR TITLE
Feature/improve posts read all

### PR DESCRIPTION
**Pull request by en-aleksandar-stojmenovic**

### CODE REVIEW

File: Twitter/Controllers/V1/PostController.cs [hunk 1]
```diff
@@ -40,12 +42,37 @@ public async Task<ActionResult<List<Post>>> Get()
 			return Ok(await _repository.Post.FindAllPostsAsync());
 		}
 
+		/// <summary>
+		/// Returns posts with pagination.
+		/// </summary>
+		/// <returns>Returns posts with pagination.</returns>
+		/// <response code = "200">Returns a list of posts with pagination.</response>
+		/// <response code = "404">Throws an exception if posts don't exist.</response>
```
Comment (bronjaharis) [inline Twitter/Controllers/V1/PostController.cs]: If we are reading posts, there can be only 0 posts, not NotFound.

File: Twitter/Controllers/V1/PostController.cs [hunk 2]
```diff
@@ -40,12 +42,37 @@ public async Task<ActionResult<List<Post>>> Get()
...
+		[HttpGet(ApiRoutes.Post.ReadAll)]
+		[ProducesResponseType(typeof(PagedResponse<Post>), 200)]
+		[CustomExceptionFilter(typeof(PostNotFoundException), HttpStatusCode.NotFound)]
```
Comment (bronjaharis) [inline Twitter/Controllers/V1/PostController.cs]: We don't need this also

File: Twitter/Queries/PaginationQuary.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	public class PaginationQuary
+	{
+		public PaginationQuary()
+		{
+			PageNumber = 1;
+			PageSize = 100;
+			CategoryId = null;
+		}
+
+		public PaginationQuary(int pageNumber, int pageSize, Guid categoryId)
```
Comment (bronjaharis) [inline Twitter/Queries/PaginationQuary.cs]: Please fix typo Query

File: Twitter/Repositories/PostRepository.cs [hunk 4]
```diff
@@ -121,5 +121,26 @@ public async Task<List<Post>> FindUserPostsAsync(Guid userId)
...
+				return await FindAll().ToListAsync();
+			}
+
+			var skip = (paginationFilter.PageNumber - 1) * paginationFilter.PageSize;
+
+			var posts = await FindByCondition(post =>
+						post.CategoryId.Equals(paginationFilter.CategoryId) && post.IsArchived == false)
+						.Skip(skip).Take(paginationFilter.PageSize).ToListAsync();
+
+			if (!posts.Any())
```
Comment (bronjaharis) [inline Twitter/Repositories/PostRepository.cs]: You can return 0 posts, it's fine.

File: Twitter/Controllers/V1/PostController.cs (hunk 1 above)
Comment (en-aleksandar-stojmenovic) [reply Twitter/Controllers/V1/PostController.cs]: I've fixed it

File: Twitter/Controllers/V1/PostController.cs (hunk 2 above)
Comment (en-aleksandar-stojmenovic) [reply Twitter/Controllers/V1/PostController.cs]: I've fixed it

File: Twitter/Queries/PaginationQuary.cs (hunk 3 above)
Comment (en-aleksandar-stojmenovic) [reply Twitter/Queries/PaginationQuary.cs]: I've fixed it

File: Twitter/Repositories/PostRepository.cs (hunk 4 above)
Comment (en-aleksandar-stojmenovic) [reply Twitter/Repositories/PostRepository.cs]: I've fixed it

**Review by bronjaharis — APPROVED** [review]

**Pull request merged**